services/tenant-ui/frontend/src/store/tenantStore.ts

```diff
@@ -335,6 +351,42 @@ export const useTenantStore = defineStore('tenant', () => {
     }
   }
 
+  function setTenantLoginDataFromLocalStorage() {
+    console.log('> setTenantFromLocalStorage');
+    try {
+      const savedTenant = localStorage.getItem('tenant');
+      tenant.value = savedTenant ? JSON.parse(savedTenant) : null;
+
+      const savedTenantConfig = localStorage.getItem('tenantConfig');
+      tenantConfig.value = savedTenantConfig
+        ? JSON.parse(savedTenantConfig)
+        : null;
+
+      const savedTaa = localStorage.getItem('tenantTaa');
+      taa.value = savedTaa ? JSON.parse(savedTaa) : null;
+
+      const savedEndorserInfo = localStorage.getItem('tenantEndorserInfo');
+      endorserInfo.value = savedEndorserInfo
+        ? JSON.parse(savedEndorserInfo)
+        : null;
+
+      const savedEndorserConnection = localStorage.getItem(
+        'tenantEndorserConnection'
+      );
+      endorserConnection.value = savedEndorserConnection
+        ? JSON.parse(savedEndorserConnection)
+        : null;
+
+      const savedPublicDid = localStorage.getItem('tenantPublicDid');
+      publicDid.value = savedPublicDid ? JSON.parse(savedPublicDid) : null;
+    } catch (err) {
+      console.info('Error processing tenant info from local storage');
+      console.error(err);
+    }
+
+    console.log('< setTenantFromLocalStorage');
+  }
+
```

Comment:
As probably noticed there's a series of calls to ACA-Py that are needed when logging in to get required tenant "profile" information which these come from.

I'm not sure we'd want to save all these to local storage though? My initial thought were it would be best to just store the **token** in local storage. If that's there then on refresh then we don't need to go through the login form, but would fetch these things on app mount/view load/wherever. So more stateless other than the token.

That probably would required some refactoring to have those calls be made outside of the LoginForm.vue. I hadn't thought through entirely what's the best pattern for this, but I think storing them could cause some issues like if something is changed in your config on the backend and you come back to the app later you'd be picking up older configuration information.

Reply:
Yeah it's nice to not have the (multiple!) calls for profile things when you refresh. 
Only thing I guess is they'd be local storage sitting in the browser and readable to someone later on whereas the token eventually expires so it's limited attack vector.
But as you said a session timer would be good, and yes I don't think anything being stored isn't public enough anyway.

So yeah we could try it out this way I think.
Note the `isIssuer` flag is used a number of places around the app and it's dependent on some of this profile info so we'd have to check out that works in this paradigm with refreshes even if you're not on a page that explicitly loads any of these.